timelock-server/src/main/java/com/palantir/atlasdb/timelock/config/TimeLockConfigMigrator.java

```diff
@@ -24,34 +23,54 @@ private TimeLockConfigMigrator() { /* Utility Class */ }
+        TimeLockInstallConfiguration install;
+        TimeLockRuntimeConfiguration runtime;
+        if (PaxosConfiguration.class.isInstance(config.algorithm())) {
+            PaxosConfiguration paxos = (PaxosConfiguration) config.algorithm();
+            install = ImmutableTimeLockInstallConfiguration.builder()
+                    .optionalPaxosConfig(ImmutablePaxosInstallConfiguration.builder()
+                            .dataDirectory(paxos.paxosDataDir())
+                            .build())
+                    .cluster(ImmutableClusterConfiguration.builder()
+                            .cluster(ServiceConfiguration.builder()
+                                    .security(paxos.sslConfiguration())
+                                    .uris(config.cluster().servers())
+                                    .build())
+                            .localServer(config.cluster().localServer())
+                            .build())
+                    .asyncLock(config.asyncLockConfiguration())
+                    .build();
 
-        TimeLockInstallConfiguration install = ImmutableTimeLockInstallConfiguration.builder()
-                .paxos(ImmutablePaxosInstallConfiguration.builder()
-                        .dataDirectory(paxos.paxosDataDir())
-                        .build())
-                .cluster(ImmutableClusterConfiguration.builder()
-                        .cluster(ServiceConfiguration.builder()
-                                .security(paxos.sslConfiguration())
-                                .uris(config.cluster().servers())
-                                .build())
-                        .localServer(config.cluster().localServer())
-                        .build())
-                .asyncLock(config.asyncLockConfiguration())
-                .build();
+            runtime = ImmutableTimeLockRuntimeConfiguration.builder()
+                    .optionalPaxos(ImmutablePaxosRuntimeConfiguration.builder()
+                            .leaderPingResponseWaitMs(paxos.leaderPingResponseWaitMs())
+                            .maximumWaitBeforeProposalMs(paxos.maximumWaitBeforeProposalMs())
+                            .pingRateMs(paxos.pingRateMs())
+                            .build())
+                    .slowLockLogTriggerMillis(config.slowLockLogTriggerMillis())
+                    .build();
+        } else if (TimestampBoundStoreConfiguration.class.isInstance(config.algorithm())) {
+            TimestampBoundStoreConfiguration timestampBoundStoreConfiguration =
+                    (TimestampBoundStoreConfiguration) config.algorithm();
+            install = ImmutableTimeLockInstallConfiguration.builder()
+                    .cluster(ImmutableClusterConfiguration.builder()
+                            .cluster(ServiceConfiguration.builder()
+                                    .security(timestampBoundStoreConfiguration.sslConfiguration())
```

Comment:
Hmm. I guess we don't have the SSL configuration elsewhere, but this does introduce a requirement that the keystore/truststore we use for the KVS and the TimeLock servers must be the same. This is probably okay, but would like to flag this.